### PR TITLE
Metrics calculator

### PR DESCRIPTION
**Pull request by dmkwis**
Code might be bad, we can fix it later :). Please parse it with your eyes and tell me if it makes sense :smiley:

### CODE REVIEW

**Comment by basiekjusz** [comment]
I've looked through all the changes, looks good to me, but I'll try to use your code in the evening.

File: eval_metrics.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    test: list[pd.DataFrame]
+    split: list[pd.DataFrame]
+
+
+def eval_model(model: AbstractLabelModel, parsed_dataset: ParsedDataset):
+    result = {"train": {}, "test": {}, "eval": {}}
+    for split_name, split in parsed_dataset.items():
+        count_cos_sim = []
+        for subgroup in split:
+            true_label = subgroup["category"][0]
```
Comment (basiekjusz) [inline eval_metrics.py:20]: Should we use `category` as `true_label`? For example for vec2text our `generated_label` will be a longer text, and therefore cos similarity might achieve poor score.

File: eval_metrics.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                true_label
+            ) @ common_utils.embedder.encode(generated_label)
+            count_cos_sim.append(cos_sim)
+        assert len(count_cos_sim) > 0, f"Length of {split_name} is 0"
+        average_cos_sim = sum(count_cos_sim) / len(count_cos_sim)
+        result[split_name]["avg_cos_sim"] = average_cos_sim
+    return result
+
+
+def read_diseases_dataset() -> ParsedDataset:
```
Comment (basiekjusz) [inline eval_metrics.py]: I guess all datasets will look the same, so maybe let's implement a generic `read_dataset(dataset_name: str)` method instead

File: create_split.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+random.seed(42_2137)
+
+
+class DatasetSplit(TypedDict):
+    train: Set[str]
+    test: Set[str]
+    eval: Set[str]
+
+
+DISEASES_TRAIN: Set[str] = set(["Diseases caused by viruses"])
```
Comment (basiekjusz) [inline create_split.py:16]: If we will not use hardcoded train, test and eval sets, we might be able to make this code generic to split any future dataset.

File: create_split.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        end_idx = min(max_idx, beg_idx + random.randint(2, 3))
+        while beg_idx < end_idx and end_idx <= max_idx:
+            subset_df = category_df.iloc[beg_idx : end_idx + 1]
+            split.append(subset_df)
+            beg_idx = end_idx + 1
+            end_idx = min(max_idx, beg_idx + random.randint(2, 3))
+    return split
+
+
+def split_diseases(dataset_dir, save_dir):
```
Comment (basiekjusz) [inline create_split.py]: I guess all the *disease* related stuff could be easily parametrized

File: abstractmodel.py
Comment (basiekjusz) [inline abstractmodel.py:1]: We should rename this file to `abstract_model.py` as it doesn't meet python's naming convention

File: eval_metrics.py (hunk 1 above)
Comment (jfpio) [reply eval_metrics.py:20]: I agree with @basiekjusz

File: create_split.py (hunk 3 above)
Comment (dmkwis) [reply create_split.py:16]: I'll make an issue about refactoring dataset related stuff, for now lets leave it like that

File: eval_metrics.py (hunk 1 above)
Comment (dmkwis) [reply eval_metrics.py:20]: I'll make an issue about that. We'll talk about it during our meeting

**Pull request merged**